pkg/compact/compact.go

```diff
@@ -743,6 +747,49 @@ func (cg *Group) areBlocksOverlapping(include *metadata.Meta, excludeDirs ...str
 	return nil
 }
 
+// isEnoughDiskSpace checks with a heuristic if there is enough disk space in the compaction group
+// in the specified directory. It adds up the space requirements of all the blocks in the compaction group
+// and then adds the smallest one again. The specified directory needs to have at least that amount of available bytes.
+// Doing this check prevents (in most cases) from downloading a lot of data and only then finding out that there is not
+// enough space.
+func (cg *Group) isEnoughDiskSpace(ctx context.Context, dir string) error {
+	du, err := diskusage.Get(dir)
+	if err != nil {
+		return errors.Wrap(err, "get disk usage")
+	}
+
+	var sumTakenByBlocks uint64
+	var smallestBlockBytes uint64 = math.MaxUint64
+
+	for blid := range cg.blocks {
+		var thisBlockBytes uint64
+		err := cg.bkt.Iter(ctx, blid.String(), func(d string) error {
+			if strings.HasSuffix(d, objstore.DirDelim) {
+				return nil
```

Comment:
I think this is wrong as compactor downloads all chunks as well, which in `chunks/` directory. This also reminds me there are no tests for this (: Do you mind adding some?

Also we might want really to be selective. E.g it might be quite confusing if someone copies `index` into same directory as `index.backup` and this will be summed here, however, compactor does not download it.  WDYT? (:

Reply:
I'm not sure how this part would skip the `chunks` directory - this is simply needed so that we would skip over things like `0123890ASDASDAS/chunks/` and only check `0123890ASDASDAS/chunks/00001`, etc.

Added some tests.

Also, I'm not sure that such a check that you are talking exists in the code at all besides the explicit handling of metadata files. AFAICT `block.Download()` is called that downloads everything that is inside there (: Could you please point me to where this check is done, @bwplotka? I feel like such a thing would be useful but probably should not be a part of this PR.

Reply:
`Iter` is not recursive. This is what I meant (: 

> FAICT block.Download() is called that downloads everything that is inside there 

You are right, let's count all then :+1: